allo/ir/builder.py

```diff
@@ -64,6 +65,9 @@
 from ..utils import get_mlir_dtype_from_str
 from ..logging import print_error_message
 
+if os.getenv("USE_AIE_MLIR_BUILDER") == "1":
+    from ..backend.experimental.external_kernel import ExternalModule
```

Comment:
Just to understand, what will happen if these guards are not added? It does not seem to be a good practice to guard it everywhere